validator/client/beacon-api/propose_attestation.go

```diff
@@ -37,27 +40,57 @@ func (c *beaconApiValidatorClient) proposeAttestation(ctx context.Context, attes
 	return &ethpb.AttestResponse{AttestationDataRoot: attestationDataRoot[:]}, nil
 }
 
-// checkNilAttestation returns error if attestation or any field of attestation is nil.
-func checkNilAttestation(attestation *ethpb.Attestation) error {
-	if attestation == nil {
-		return errors.New("attestation is nil")
-	}
-
-	if attestation.Data == nil {
+// checkNilContentsAttestation returns error if attestation or any field of attestation is nil.
+func checkNilContentsAttestation(attestation ethpb.Att) error {
+	if attestation.GetData() == nil {
```

Comment:
I see you've changed this to be used in the test but wouldn't it be fine to leave it as it was so as `checkNilAttestation`? Since it's doing all nil checks this way instead of separating it

Reply:
It doesn't work on checking `if attestation == nil` even when it's nil it shows up as not nil on the interface check.